src/state/StatePath.java

```diff
@@ -33,6 +33,10 @@ public boolean isEmpty() {
         return path.isEmpty();
     }
 
+    public boolean atEnd() {
```

Comment:
Unit tests for new public method?

Reply:
I don't think there's anything valuable we can unit test for this function by itself, so I've instead added asserts for `atEnd` in other `StatePath` unit tests for sanity checking.

Reply:
The case where the position is in the middle of the path is never tested.